intent_classification/cnn/train_with_pretrained_embeddings.py

```diff
@@ -0,0 +1,100 @@
+import tensorflow as tf
+import math
+import sys
+sys.path.append("../")
+import atis_data
+
+FLAGS = tf.app.flags.FLAGS
+tf.app.flags.DEFINE_string("data_dir", '../../ATIS', "Data directory")
+tf.app.flags.DEFINE_integer("in_vocab_size", 10000, "max vocab Size.")
+tf.app.flags.DEFINE_string("max_in_seq_len", 30, "max in seq length")
+tf.app.flags.DEFINE_integer("max_data_size", 10000, "max training data size")
+tf.app.flags.DEFINE_integer("batch_size", 100, "batch size")
+tf.app.flags.DEFINE_integer("iterations", 10000, "number of iterations")
+tf.app.flags.DEFINE_integer("embedding_size", 300, "size of embedding")
+
+def train():
+	atis = atis_data.AtisData(FLAGS.data_dir, FLAGS.in_vocab_size,
+			 FLAGS.max_in_seq_len, FLAGS.max_data_size, FLAGS.embedding_size)
+
+	# define parameters of cnn
+	num_filters = 32
+	filter_sizes = [2, 3, 4]
+	num_filters_total = num_filters * len(filter_sizes)
+	dropout_keep_prob = 0.4
+	normal_initializer = tf.random_normal_initializer(stddev=0.1)
+
+	#get data
+	num_classes = atis.get_number_of_labels()
+	l2_loss = tf.constant(0.0)
+
+	train_x = tf.placeholder(tf.float32, [None, FLAGS.max_in_seq_len, FLAGS.embedding_size])
+	# [None,sentence_length,embed_size]
+	train_y = tf.placeholder(tf.float32, [None, num_classes])
+	embedded_expanded = tf.expand_dims(train_x, -1)
+	# [None,sentence_length, feature_size, 1)
+	
+	W_projection = tf.get_variable("W_projection",
+			shape=[num_filters_total, num_classes],
+			initializer=normal_initializer) #[feature _size,label_size]
+	b_projection = tf.get_variable("b_projection",shape=[num_classes]) #[label_size] 
+
+	pooled_outputs = []
+	for i,filter_size in enumerate(filter_sizes):
+		with tf.name_scope("convolution-pooling-%s" %filter_size):
+			filter = tf.get_variable("filter-%s"%filter_size,
+						[filter_size,FLAGS.embedding_size,1,num_filters],
```

Comment:
I'm just curious - is there a reason why one of the dimension of the filter is `FLAGS.embedding_size`, can it be smaller than that? 

In normal images, filters are usually much smaller than the h x w dimension of the image. 

I'm asking because today you drew the filters this way as well, but I thought that was just by chance - but it looks like there is a reason behind it?

Reply:
It can't be smaller. Read http://www.wildml.com/2015/11/understanding-convolutional-neural-networks-for-nlp/